### PR TITLE
Add CI Nightly testnet deployment job

### PR DESCRIPTION
**Pull request by O1ahmad**
Adds a nightly deployment CI job for launching a long-running testnet based on compatible for daily evaluation and refactors the `ci-net` CI deployment deploy process/specification to support pre-deployment steps and deployment conditionals.

Also add installation of the gcloud sdk into daemon containers for `gs-util` use (executed when uploading network block data to gcloud) and bump helm chart versions to latest.

**Test:** Buildkite CI

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: dockerfiles/Dockerfile-coda-daemon [hunk 1]
```diff
@@ -24,7 +28,8 @@ RUN apt-get -y update && \
     apt-transport-https \
     ca-certificates \
     dnsutils \
-    tzdata && \
+    tzdata  \
+    google-cloud-sdk && \
```
Comment (0x0I) [inline dockerfiles/Dockerfile-coda-daemon:32]: Ok to add to daemon image? (..or maybe too heavy to add for general use)

File: buildkite/src/Jobs/Test/Nightly.dhall [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+let Pipeline = ../../Pipeline/Dsl.dhall
+
+let DeployTestnet = ../../Command/DeployTestnet.dhall
+
+
+let spec = DeployTestnet.DeploySpec::{
+  testnetLabel = "nightly",
+  workspace = "nightly",
+  deployCondition = "build.branch == 'compatible' && build.env('NIGHTLY') == 'true'",
+  preDeploy = "terraform destroy -auto-approve && cp /tmp/gcp_creds.json ../../../gcloud-keyfile.json",
```
Comment (0x0I) [inline buildkite/src/Jobs/Test/Nightly.dhall:13]: maybe add a post deploy operation for removing (or archiving) existing network block data?

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Comment by mrmr1993** [comment]
Stale and superseded; closing.

**Pull request closed without merging**